lpc55_sign/src/bin/lpc55_sign.rs

```diff
@@ -137,15 +157,293 @@ fn main() -> Result<()> {
                 &dest_bin, &dest_cmpa
             );
         }
-        ImageType::EccImage {
+        Command::EccImage {
             src_bin,
             priv_key,
             dest_bin,
         } => {
             sign_ecc::ecc_sign_image(&src_bin, &priv_key, &dest_bin)?;
             println!("Done! ECC image written to {:?}", &dest_bin);
         }
+        Command::VerifySignedImage {
+            src_cmpa,
+            src_cfpa,
+            src_img,
+            verbose,
+        } => {
+            println!("=== CMPA ====");
+            let cmpa = {
+                let mut cmpa_bytes = [0u8; 512];
+                let mut cmpa_file = std::fs::File::open(src_cmpa)?;
+                cmpa_file.read_exact(&mut cmpa_bytes)?;
+                CMPAPage::from_bytes(&cmpa_bytes)?
```

Comment:
(iirc the `from_bytes` errors are also crappy, which would be a nice place to use `context` as well, but hitting this is way less common -- it basically requires a corrupt or invalid file, instead of merely a typo)

Reply:
Fixed!